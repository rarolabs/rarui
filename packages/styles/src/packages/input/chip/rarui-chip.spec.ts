import { chip } from ".";

describe("Chip Styles", () => {
  it("should apply base styles correctly", () => {
    const className = chip.classnames.chip();
    expect(className).toContain("chip");
  });

  it("should apply size variants correctly", () => {
    const sizes = ["medium", "small"] as const;
    sizes.forEach((size) => {
      const className = chip.classnames.chip({ size });
      expect(className).toContain(size);
      expect(className).toMatchSnapshot(`size-${size}`);
    });
  });

  it("should apply pill variant styles correctly", () => {
    const className = chip.classnames.chip({ pill: true });
    expect(className).toContain("pill");
    expect(className).toMatchSnapshot("pill");
  });

  it("should apply selected variant styles correctly", () => {
    const className = chip.classnames.chip({ selected: true });
    expect(className).toContain("selected");
    expect(className).toMatchSnapshot("selected");
  });

  it("should apply textOverflow variant styles correctly", () => {
    const className = chip.classnames.chip({ textOverflow: true });
    expect(className).toContain("textOverflow");
    expect(className).toMatchSnapshot("textOverflow");
  });

  it("should apply content styles correctly", () => {
    const className = chip.classnames.content({ textOverflow: true });
    expect(className).toContain("content");
    expect(className).toMatchSnapshot("content");
  });

  it("should apply close button styles correctly", () => {
    const className = chip.classnames.close;
    expect(className).toMatchSnapshot("close");
  });

  it("should apply overlay styles correctly", () => {
    const className = chip.classnames.overlay;
    expect(className).toMatchSnapshot("overlay");
  });
});
