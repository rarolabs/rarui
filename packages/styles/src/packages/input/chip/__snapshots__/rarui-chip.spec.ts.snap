// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Styles should apply close button styles correctly: close 1`] = `"rarui-chip_close__1rk47pp8"`;

exports[`Chip Styles should apply content styles correctly: content 1`] = `"rarui-chip_content__1rk47pp6 rarui-chip_content_textOverflow_true__1rk47pp7"`;

exports[`Chip Styles should apply overlay styles correctly: overlay 1`] = `"rarui-chip_overlay__1rk47pp9"`;

exports[`Chip Styles should apply pill variant styles correctly: pill 1`] = `"rarui-chip__1rk47pp0 rarui-chip_pill_true__1rk47pp1"`;

exports[`Chip Styles should apply selected variant styles correctly: selected 1`] = `"rarui-chip__1rk47pp0 rarui-chip_selected_true__1rk47pp2"`;

exports[`Chip Styles should apply size variants correctly: size-medium 1`] = `"rarui-chip__1rk47pp0 rarui-chip_size_medium__1rk47pp3"`;

exports[`Chip Styles should apply size variants correctly: size-small 1`] = `"rarui-chip__1rk47pp0 rarui-chip_size_small__1rk47pp4"`;

exports[`Chip Styles should apply textOverflow variant styles correctly: textOverflow 1`] = `"rarui-chip__1rk47pp0 rarui-chip_textOverflow_true__1rk47pp5"`;
